### PR TITLE
Pre evaluate outer and Extends new evaluation

### PR DESCRIPTION
**Pull request by iusildra**

### CODE REVIEW

File: modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/evaluator/Validation.scala [hunk 1]
```diff
@@ -81,7 +81,7 @@ object Validation {
       if (value == null) Recoverable("Found null value, expected non-null value")
       else Valid(value)
     } catch {
-      case t: Throwable => Fatal(new Exception(t))
+      case t: Throwable => CompilerRecoverable(new Exception(t))
```
Comment (adpi2) [inline modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/evaluator/Validation.scala:84]: If something unexpected happen, for instance a disconnection, it is probably not recoverable.

File: modules/tests/src/test/scala/ch/epfl/scala/debugadapter/internal/RuntimeEvaluatorTests.scala [hunk 2]
```diff
@@ -616,6 +616,54 @@ abstract class RuntimeEvaluatorTests(val scalaVersion: ScalaVersion) extends Deb
...
+         |    val aAAaaa1 = new aAA.AAA(42)
+         |    val aAAaaa2 = new aAA.AAA(43)
+         |    println("ok")
+         |  }
+         |}
+         |""".stripMargin
+    implicit val debuggee = TestingDebuggee.mainClass(source, "example.Main", scalaVersion)
+    check(
+      Breakpoint(28),
+      Evaluation.success("new a.AA") { res => res.startsWith("A$AA@") },
```
Comment (adpi2) [inline modules/tests/src/test/scala/ch/epfl/scala/debugadapter/internal/RuntimeEvaluatorTests.scala:662]: If we stop in `A` (`this` is of type `A`) can we evaluate `new AA` ?

File: modules/tests/src/test/scala/ch/epfl/scala/debugadapter/internal/RuntimeEvaluatorTests.scala [hunk 3]
```diff
@@ -616,6 +616,54 @@ abstract class RuntimeEvaluatorTests(val scalaVersion: ScalaVersion) extends Deb
       Evaluation.success("test(new Integer(1))", "boxed int")
     )
   }
+
+  test("Should instantiate inner classes") {
```
Comment (adpi2) [inline modules/tests/src/test/scala/ch/epfl/scala/debugadapter/internal/RuntimeEvaluatorTests.scala:626]: Does it work also with Java classes ?

File: modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/evaluator/Validation.scala (hunk 1 above)
Comment (iusildra) [reply modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/evaluator/Validation.scala:84]: Marked it as an issue for a next PR

File: modules/tests/src/test/scala/ch/epfl/scala/debugadapter/internal/RuntimeEvaluatorTests.scala (hunk 2 above)
Comment (iusildra) [reply modules/tests/src/test/scala/ch/epfl/scala/debugadapter/internal/RuntimeEvaluatorTests.scala:662]: Nop :upside_down_face:

File: modules/tests/src/test/scala/ch/epfl/scala/debugadapter/internal/RuntimeEvaluatorTests.scala (hunk 3 above)
Comment (iusildra) [reply modules/tests/src/test/scala/ch/epfl/scala/debugadapter/internal/RuntimeEvaluatorTests.scala:626]: Now it should

**Pull request merged**